tests/unit_tests/include/test_helper.hpp

```diff
@@ -60,24 +67,34 @@
 class DeviceNamePrint {
 public:
     std::string operator()(testing::TestParamInfo<cl::sycl::device> dev) const {
-        if (dev.param.is_cpu())
-            return std::string("CPU");
-        if (dev.param.is_host())
-            return std::string("HOST");
-        if (dev.param.is_gpu()) {
-            unsigned int vendor_id =
-                static_cast<unsigned int>(dev.param.get_info<cl::sycl::info::device::vendor_id>());
-            switch (vendor_id) {
-                case INTEL_ID:
-                    return std::string("INTELGPU");
-                case NVIDIA_ID:
-                    return std::string("NVIDIAGPU");
-            }
+        std::string dev_name = dev.param.get_info<cl::sycl::info::device::name>();
+        for (std::string::size_type i = 0; i < dev_name.size(); ++i) {
+            if (!isalnum(dev_name[i]))
+                dev_name[i] = '_';
         }
-        if (dev.param.is_accelerator())
-            return std::string("ACCELERATOR");
-        return std::string("UNKNOWN");
+        return dev_name;
     }
 };
 
+/* to accomodate Windows and Linux differences between alligned_alloc and
+   _aligned_malloc calls use onemkl::aligned_alloc and onemkl::aligned_free instead */
+namespace onemkl {
+
+static inline void *aligned_alloc(size_t align, size_t size) {
+#ifdef _WIN64
+    return ::_aligned_malloc(size, align);
+#else
+    return ::aligned_alloc(align, size);
+#endif
+}
+
+static inline void aligned_free(void *p) {
+#ifdef _WIN64
+    ::_aligned_free(p);
+#else
+    ::free(p);
+#endif
+}
+} // namespace onemkl
+
```

Comment:
This seems like duplicate with src/include/allocation_helper.hpp. Is there any way to avoid duplication?

Reply:
That was done intentionally, as we didn't want make these functions public. In src we need them to build libraries, in testing - for testing. We don't use includes from src in testing and vice versa - we don't use includes from testing in src.

Reply:
Thanks for the explanation.